lib/Driver/DarwinToolChains.cpp

```diff
@@ -84,22 +84,29 @@ toolchains::Darwin::constructInvocation(const InterpretJobAction &job,
 }
 
 static StringRef
-getDarwinLibraryNameSuffixForTriple(const llvm::Triple &triple) {
+getDarwinLibraryNameSuffixForTriple(const llvm::Triple &triple,
+                                    bool distinguishSimulator = true) {
   switch (getDarwinPlatformKind(triple)) {
   case DarwinPlatformKind::MacOS:
     return "osx";
+  case DarwinPlatformKind::IPhoneOSSimulator:
+    if (distinguishSimulator)
+      return "iossim";
+    LLVM_FALLTHROUGH;
   case DarwinPlatformKind::IPhoneOS:
     return "ios";
```

Comment:
What you *really* want here IMO is Swift's switch on tuple, so you could switch on both.
Failing that, it would be nice to make the symmetry wrt `distinguishSimulator` more apparent.
E.g.: 
```
DarwinPlatformKind mapSimulatorToOS(DarwinPlatformKind kind) {
switch kind {
  case DarwinPlatformKind::MacOS: return DarwinPlatformKind::MacOS
  case IPhoneOSSimulator: return IPhoneOS
  case IPhoneOS: return IPhoneOS
...
}

static StringRef
getDarwinLibraryNameSuffixForTriple(const llvm::Triple &triple,
                                    bool distinguishSimulator = true) {
  const DarwinPlatformKind kind = getDarwinPlatformKind(triple)
  const DarwinPlatformKind effectiveKind = distinguishSimulator ? kind : mapSimulatorToOS(kind)
  switch (effectiveKind) {
  case DarwinPlatformKind::MacOS:
    return "osx";
  case DarwinPlatformKind::IPhoneOSSimulator:
    return "iossim";
  case DarwinPlatformKind::IPhoneOS:
    return "ios";
   ...
  }
````
I like this because IMO it reduces the possibility for future errors.